src/Command/Email/ConfigurePlatformEmailCommand.php

```diff
@@ -0,0 +1,331 @@
+          // For any other error from the API, the setup will exit.
+          if (strpos($e, 'is already enabled on this environment') === FALSE) {
+            $this->io->error($e->getMessage());
+            return FALSE;
+          }
+          else {
+            $this->io->warning($env->label . ' - ' . $e->getMessage());
+          }
+        }
+      }
+    }
+    return TRUE;
+  }
+
+  /**
+   * Validates the URL entered as the base domain name.
+   *
+   * @param string $url
+   *
+   * @return string
+   * @throws \Symfony\Component\Validator\Exception\ValidatorException
+   */
+  public static function validateUrl(string $url): string {
+    $violations = Validation::createValidator()->validate($url, [
+      new NotBlank(),
+    ]);
+    if (count($violations)) {
+      throw new ValidatorException($violations->get(0)->getMessage());
+    }
+    return $url;
+  }
+
+  /**
+   * Retrieves a domain registration UUID given the domain name.
+   *
+   * @param \AcquiaCloudApi\Connector\Client $client
+   * @param SubscriptionResponse $subscription
+   * @param string $base_domain
+   *
+   * @return mixed
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function fetchDomainUuid(Client $client, $subscription, $base_domain) {
+    $domains_response = $client->request('get', "/subscriptions/{$subscription->uuid}/domains");
+    foreach ($domains_response as $domain) {
+      if ($domain->domain_name === $base_domain) {
+        return $domain->uuid;
+      }
+    }
+    throw new AcquiaCliException("Could not find domain $base_domain");
+  }
+
+  /**
+   * Creates a TXT file, either in JSON or YAML format,
+   * of the DNS records needed to complete Platform Email setup.
+   *
+   * @param \AcquiaCloudApi\Connector\Client $client
+   * @param SubscriptionResponse $subscription
+   * @param string $domain_uuid
+   * @param string $file_format
+   *
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function createDnsText(Client $client, $subscription, $domain_uuid, $file_format): void {
+    $domain_registration_response = $client->request('get', "/subscriptions/{$subscription->uuid}/domains/{$domain_uuid}");
+    if (!isset($domain_registration_response->dns_records)) {
+      throw new AcquiaCliException('Could not retrieve DNS records for this domain. Please try again by rerunning this script with the domain that you just registered.');
+    }
+    $records = [];
+    $this->localMachineHelper->getFilesystem()->remove('dns-records.txt');
+    if ($file_format === 'JSON') {
+      foreach ($domain_registration_response->dns_records as $record) {
+        unset($record->health);
+        $records[] = $record;
+      }
+      $this->logger->debug(json_encode($records));
+      $this->localMachineHelper->getFilesystem()
+            ->dumpFile('dns-records.txt', json_encode($records, JSON_PRETTY_PRINT));
```

Comment:
seems like the filename should use a yaml or json extension depending on the format.